### ISSUE TITLE
Peek commands trigger folding again after document opened

### ISSUE THREAD

**Issue by eMuonTau** [eramdam/fold-imports#1]
Hi,
First i want to thank you for this awesome extension. 

I found out hovering over code while CTRL is pressed folds imports again. Same happens if you execute peek commands (eg: PeekDefinition) with shortcut or from context menu. I think peek commands trigger `onDidOpenTextDocument` too. Please let me know if you need more info.

Language: Typescript
fold-imports: 1.0.1
VSCode: 1.45.1
Node: v12.16.3
OS: Windows 10 1909 x64
Extension List: All extensions are disabled

**Comment by eramdam** [comment]
I think I managed to repro this only once and then not again...
- I peeked on an import (using the command from the palette)
- I was moved somewhere (cant tell if it was to the corresponding file or if I stayed on the same file)
- the imports collapsed

Does that sound similar to what you're experiencing?

**Comment by eMuonTau** [comment]
I tried it on another machine with Windows 7 but it is same. Every time i hover on particular tokens with CTRL pressed, it is trying to fold imports even if it is already folded. It happens on most of the imported types. It also happens on some object properties and functions but i couldn't find a pattern. 

Another issue is when it fold the imports, it scrolls back to where the cursor is located. 
- Place the cursor at the end of a long file 
- Hover on some imported type definitions ()
- Imports will be folded 
- Editor will be scrolled to end of the file where cursor is located

**Comment by eramdam** [comment]
Okay I think I managed to reproduce the CTRL thing. On macOS it's hover + Alt and indeed the imports get folded in the current file no matter what I do. I'll need to look into this.

**Comment by eMuonTau** [comment]
I just saw your message. I sent a PR that executes fold command only if document is active one. I don't know if there is a better way to fix it. I did not work on a vscode extension for months.

**Comment by eramdam** [comment]
@eMuonTau I didn't see your message until now and opened a PR https://github.com/eramdam/fold-imports/pull/3.

I feel more comfortable with my solution than with a `setImmediate` whose timing might change in the future (this already happened in the past).

**Comment by eramdam** [comment]
@eMuonTau PR merged and published under 1.0.2. Let me know how it goes but as far as I can tell it _should_ have fixed the issue you reported 🙇‍♂️

**Comment by eMuonTau** [comment]
@eramdam Thank you for fixing it so fast. No more teleports in my long test files :sweat_smile: It works perfect.

I think you used `document.fileName.includes` instead of equality for a reason. I don't know why but i see an extra `.git` extension in the `document.fileName`. Active editor file name ends with `.ts` but newly opened file name ends with `.ts.git`.

**Comment by eramdam** [comment]
> @eramdam Thank you for fixing it so fast. No more teleports in my long test files 😅 It works perfect.
> 
> I think you used `document.fileName.includes` instead of equality for a reason. I don't know why but i see an extra `.git` extension in the `document.fileName`. Active editor file name ends with `.ts` but newly opened file name ends with `.ts.git`.

Yes that's exactly the reason. I'm not a fan of this solution but I couldn't find any other "clean" way to compare files 🙈

**Comment by eMuonTau** [comment]
This issue popped up again after 1.0.3 release. Should i open a new issue?

**Comment by eramdam** [comment]
> This issue popped up again after 1.0.3 release. Should i open a new issue?

I noticed that as well, really not sure why. I'll re-open this issue 😞

**Comment by eramdam** [comment]
I opened https://github.com/eramdam/fold-imports/pull/7 to fix the issue. I'll run it today and publish it publicly if it works well. @eMuonTau I can make a build if you want to test it out locally before it goes out.

**Comment by eramdam** [comment]
Released 1.0.4 with the fix, let me know how this goes.

**Comment by eMuonTau** [comment]
I installed 1.0.4 and it solved the problem. Thanks.
